de.dlr.sc.virsat.server.test/src/de/dlr/sc/virsat/server/resources/modelaccess/QudvResourceTest.java

```diff
@@ -0,0 +1,298 @@
+		assertTrue(obj instanceof SimpleUnit);
+	}
+	
+	@Test
+	public void testUnitDelete() {
+		int unitCount = units.size();
+		Response response = getTestRequestBuilder(ModelAccessResource.QUDV + "/" + QudvResource.UNIT + "/" + unitNotDerived.getUuid().toString()).delete();
+		
+		assertEquals(HttpStatus.OK_200, response.getStatus());
+		assertEquals("One unit removed", unitCount - 1, unitManagement.getSystemOfUnit().getUnit().size());
+	}
+	
+	@Test
+	public void testQuantityKindGet() throws Exception {
+		IBeanUuid quantityKindBean = new BeanQuantityKindFactory().getInstanceFor(qkNotDerived);
+		testGet(quantityKindBean, ModelAccessResource.QUDV + "/" + QudvResource.QUANTITY_KIND, new Class[] {quantityKindBean.getClass()});
+	}
+	
+	@Test
+	public void testQuantityKindPut() throws JAXBException {
+		IBeanQuantityKind<? extends AQuantityKind> quantityKindBean = new BeanQuantityKindFactory().getInstanceFor(qkNotDerived);
+		
+		Response response = getTestRequestBuilder(ModelAccessResource.QUDV + "/" + QudvResource.QUANTITY_KIND)
+				.put(Entity.entity((BeanQuantityKindSimple) quantityKindBean, MediaType.APPLICATION_JSON_TYPE));
+		
+		assertEquals(HttpStatus.OK_200, response.getStatus());
+	}
+	
+	@Test
+	public void testQuantityKindCreate() {
+		int qkCount = quantityKinds.size();
+		Response response = getTestRequestBuilderWithQueryParam(
+				ModelAccessResource.QUDV + "/" + QudvResource.QUANTITY_KIND, ModelAccessResource.QP_NAME, BeanQuantityKindSimple.class.getSimpleName())
+				.post(Entity.json(null));
+		
+		assertEquals(HttpStatus.OK_200, response.getStatus());
+		assertEquals("One new qk added", qkCount + 1, quantityKinds.size());
+		
+		String uuid = response.readEntity(String.class);
+		AQuantityKind obj = RepositoryUtility.findQuantityKind(uuid, resourceSet.getRepository());
+		assertNotNull(obj);
+		assertThat((List<AQuantityKind>) quantityKinds, hasItem(obj));
+		assertTrue(obj instanceof SimpleQuantityKind);
+	}
+	
+	@Test
+	public void testQuantityKindDelete() {
+		int qkCount = quantityKinds.size();
+		Response response = getTestRequestBuilder(ModelAccessResource.QUDV + "/" + QudvResource.QUANTITY_KIND + "/" + qkNoReference.getUuid().toString()).delete();
+		
+		assertEquals(HttpStatus.OK_200, response.getStatus());
+		assertEquals("One quantity kind removed", qkCount - 1, quantityKinds.size());
+	}
+	
+	@Test
+	public void testPrefixGet() throws Exception {
+		IBeanUuid prefixBean = new BeanPrefix(prefix);
+		testGet(prefixBean, ModelAccessResource.QUDV + "/" + QudvResource.PREFIX, new Class[] {prefixBean.getClass()});
+	}
+	
+	@Test
+	public void testPrefixPut() throws JAXBException {
+		IBeanUuid prefixBean = new BeanPrefix(prefix);
+		
+		Response response = getTestRequestBuilder(ModelAccessResource.QUDV + "/" + QudvResource.PREFIX)
+				.put(Entity.entity(prefixBean, MediaType.APPLICATION_JSON_TYPE));
+		
+		assertEquals(HttpStatus.OK_200, response.getStatus());
+	}
+	
+	@Test
+	public void testPrefixCreate() {
```

Comment:
You are reusing prefix count  and prefix count does it make sense to define it in setup ?